### PR TITLE
port global chanops and logging of kicks from sigmabot

### PR DESCRIPTION
**Pull request by MacFan4000**

### CODE REVIEW

File: sopel_channelmgnt/channelmgnt/__init__.py [hunk 1]
```diff
@@ -35,6 +42,9 @@ def configure(config):
     config.define_section('channelmgnt', ChannelmgntSection, validate=False)
     config.channelmgnt.configure_setting('datafile', 'Where is the datafile for channelmgnt?')
     config.channelmgnt.configure_setting('support_channel', 'What channel should users ask for help in?')
+    config.channelmgnt.configure_setting('global_chanops', 'If anybody, who should have chanop access in all channels? (if any then enter nickserv accounts)')
```
Comment (RhinosF1) [inline sopel_channelmgnt/channelmgnt/__init__.py]: We support global_chanops now, it's called inherits from

File: sopel_channelmgnt/channelmgnt/__init__.py [hunk 2]
```diff
@@ -35,6 +42,9 @@ def configure(config):
...
+    config.channelmgnt.configure_setting('log_kicks', 'Should kicks be logged to a channel? (true/false)')
```
Comment (RhinosF1) [inline sopel_channelmgnt/channelmgnt/__init__.py]: use channel specific config formats in datafile

File: sopel_channelmgnt/channelmgnt/__init__.py [hunk 3]
```diff
@@ -35,6 +42,9 @@ def configure(config):
...
+    config.channelmgnt.configure_setting('log_channel', 'If you said true to the previous question, then please provide the channel where kicks should be logged to')
```
Comment (RhinosF1) [inline sopel_channelmgnt/channelmgnt/__init__.py]: as above

**Review by RhinosF1 — CHANGES_REQUESTED** [review]
see review

File: sopel_channelmgnt/channelmgnt/__init__.py (hunk 1 above)
Comment (MacFan4000) [reply sopel_channelmgnt/channelmgnt/__init__.py]: Does having chanops and inherits from defined at the same time work?

File: sopel_channelmgnt/channelmgnt/__init__.py (hunk 3 above)
Comment (MacFan4000) [reply sopel_channelmgnt/channelmgnt/__init__.py]: ?

File: sopel_channelmgnt/channelmgnt/__init__.py (hunk 2 above)
Comment (MacFan4000) [reply sopel_channelmgnt/channelmgnt/__init__.py]: ?

File: sopel_channelmgnt/channelmgnt/__init__.py (hunk 1 above)
Comment (RhinosF1) [reply sopel_channelmgnt/channelmgnt/__init__.py]: see my other PR

File: sopel_channelmgnt/channelmgnt/__init__.py (hunk 2 above)
Comment (RhinosF1) [reply sopel_channelmgnt/channelmgnt/__init__.py]: log_kicks should be per channel overrideable

Comment (MacFan4000) [reply sopel_channelmgnt/channelmgnt/__init__.py]: I was trying to mimic sigmabot, where kicks for all channels went to the same channel

Comment (RhinosF1) [reply sopel_channelmgnt/channelmgnt/__init__.py]: I'm saying that's not ideal. We should allow channels to have their own logs too.

File: sopel_channelmgnt/channelmgnt/__init__.py (hunk 1 above)
Comment (MacFan4000) [reply sopel_channelmgnt/channelmgnt/__init__.py]: Ok so I can't add that for us until that is merged

**Review by RhinosF1 — CHANGES_REQUESTED** [review]
Blocking on the PR to update channelmgnt get chanops logic. I don't want to make the issue worse.

**Comment by MacFan4000** [comment]
> Blocking on the PR to update channelmgnt get chanops logic. I don't want to make the issue worse.

I honestly don't see how this would make it any worse.

**Comment by RhinosF1** [comment]
This PR (ab)uses the same logic

File: sopel_channelmgnt/channelmgnt/__init__.py [hunk 4]
```diff
@@ -43,6 +43,10 @@ def configure(config):
     config.channelmgnt.configure_setting('support_channel', 'What channel should users ask for help in?')
 
 
+BOLD = '\x02'
```
Comment (RhinosF1) [inline sopel_channelmgnt/channelmgnt/__init__.py]: use sopel's built in

File: sopel_channelmgnt/channelmgnt/__init__.py [hunk 5]
```diff
@@ -43,6 +43,10 @@ def configure(config):
...
+GREEN = '\x0303'
```
Comment (RhinosF1) [inline sopel_channelmgnt/channelmgnt/__init__.py]: as with BOLD

**Review by RhinosF1 — CHANGES_REQUESTED** [review]
https://sopel.chat/docs/api.html#module-sopel.formatting

**Review by RhinosF1 — COMMENTED** [review]
Do we have to increase line lengths or can lines be appropriately broke up?

**Review by RhinosF1 — APPROVED** [review]
Works for now, I'll build the new system on top later.

**Pull request merged**